File: kods_C_C++/PYTHON/sentimental-readability/readability.py
import cs50
from cs50 import get_string
import math
import string


def main():

    text = get_string("Text:")
    chars = count_characters(text)
    words = count_words(text)
    sentances = count_sentances(text)

    print(f"ch: {chars} \nwords: {words}\nsentances: {sentances}")
    get_grade(chars, words, sentances)


def get_grade(characters, words, sentances):
    L = round((characters/words) * 100)
    S = round((sentances/words) * 100)
    grade = 0.0588 * L - 0.296 * S - 15.8
    print(f"L: {L}\nS: {S}")
    grade = round(grade)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


def count_characters(text):
    text = text.lower()
    count = 0
    for i in range(len(text)):
        if ord(text[i]) >= 97 and ord(text[i]) <= 122:
            count += 1
    return count


def count_words(text):
    words = 0
    if ord(text[0]) != " ":
        words += 1

    for ch in range(len(text)):
        if text[ch] == " ":
            words += 1
    return words


def count_sentances(text):
    count = 0
    for ch in range(len(text)):
        if text[ch] in [".", "?", "!"]:
            count += 1
    return count


main()

File: kods_C_C++/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
// decelarations
int count_characters(string text);
int count_words(string text);
int count_sentance(string text);
void get_grade(double ch, double words, double sent);
int length;

int main(void)
{
    string text = get_string("TEXT: \n");

    length = strlen(text);

    int characters = count_characters(text);
    int words = count_words(text);
    int sentances = count_sentance(text);
    get_grade(characters, words, sentances);

    // printf("%i letters\n", characters);
    // printf("%i words\n", words);
    // printf("%i sentances\n", sentances);
}

int count_characters(string text)
{
    int count = 0;
    for (int i = 0; i < length; i++)
    {
        // lower case:
        if ((text[i] >= 97 && text[i] <= 122) || (text[i] >= 65 && text[i] <= 90))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int words = 0;
    if (text[0] != ' ')
    {
        words++;
    }

    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ') // space can also be assumed as 32 ASCII;
        {
            words++;
        }
    }
    return words;
}

int count_sentance(string text)
{
    int sentances = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentances++;
        }
    }
    return sentances;
}

void get_grade(double ch, double words, double sent)
{
    double grade = 0;
    // to get L:
    double L = (ch / words) * 100;
    // to get S:
    double S = (sent / words) * 100;

    // Calculate Coleman-liau Index
    grade = 0.0588 * L - 0.296 * S - 15.8;

    grade = round(grade);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
        return;
    }

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)grade);
    }
}